internal/plugin/platform.go

```diff
@@ -428,17 +429,37 @@ func (s *platformServer) Deploy(
 	internal := s.internal()
 	defer internal.Cleanup.Close()
 
+	// NOTE(izaak): Argmapper has two different ways to get a DeclaredResourcesResp:
+	// the value we provide below, or converting the pb.DeclaredResources supplied by core
+	// with a mapper function. In practice, it uses this one (which is important), but I don't
+	// know if we can rely on that behavior. The commented-out snippet below removes the other arg.
+	declaredResourcesResp := &component.DeclaredResourcesResp{}
```

Comment:
Can you go into more detail as to why there ends up being two DeclaredResources? This kind of seems like we're working around a bug or undesired behavior that gives us the wrong declared resource message in the future (based off the comment)

Reply:
Fixed - there's only one now!